### PR TITLE
mmn: CI/CD mmn binary name

### PR DESCRIPTION
**Pull request by Hikaru2035**
# Ticket
https://github.com/mezonai/mmn-tx-explorer/issues/642
#516 

# General
- [x] No unnecessary files, debug logs, or commented code included.
- [ ] PR has at least 1 peer approval before senior review.
- [x] I fully understand the code I submitted.
- [x] No hardcoded configuration or secrets.
- [ ] **Environment Variables** changes? (Leave unchecked if N/A)

# Design & Solution Clarity
- [ ] A design diagram (sequence / flow) is attached or linked.
- [ ] The implementation matches the design.
- [ ] All major flows and decision points are represented in the diagram.
- [ ] Error handling and edge cases are clearly defined in the design.

Design link (diagram / doc):
`<paste link here>`

# Testing
- [ ] Unit tests cover main logic and important edge cases.

> Note: Complex logic must not rely on manual testing only.

# UI evidence (For frontend only)
`<paste screenshots here>`

# Peer Review Checklist
- [ ] I understand the overall solution and flow.
- [ ] The code follows the provided design diagram.
- [ ] No obvious logic or security issues found.
- [ ] Tests are sufficient for the scope of this change.

### CODE REVIEW

**Comment by Hikaru2035** [comment]
Requirement:
MMN binary name must follow semantic version format: vX.Y.Z (e.g. v0.0.0)

File: .github/workflows/deploy-mmn-services.yml [hunk 1]
```diff
@@ -157,8 +152,14 @@ jobs:
           CGO_LDFLAGS: "-L$(pwd)/libs -lrocksdb -lstdc++ -lm -lz -lbz2 -lsnappy -llz4 -lzstd -luring"
         run: |
           mkdir -p build
+          TAG_NAME="${{ steps.determine_release_tag.outputs.tag }}"
```
Comment (tuananhncc) [inline .github/workflows/deploy-mmn-services.yml]: redudant 
```
TAG_NAME="${{ steps.determine_release_tag.outputs.tag }}"
 if [ -z "$TAG_NAME" ]; then TAG_NAME="v0.0.0"; fi
echo "Building version: $TAG_NAME"
echo "tag_name=${TAG_NAME}" >> $GITHUB_OUTPUT
```

**Review by tuananhncc — APPROVED** [review]
